### ISSUE TITLE
When i delete a function, the local variables do not also delete. 

### ISSUE THREAD

**Issue by jaredgib** [thunkable/thunkable-issues#478]
Thanks so much for filing an issue or feature request! Please fill out the following (wherever relevant):

The more info, the faster we will be able to address it!

### Platform
Thunkable X

### Issue/Request Origin
My own experience:
When i create a function, i may create input parameters and use those as locally scoped variables. Great! But, when i delete that function, those locally scoped variables do not disappear from the variable list. I can still call them. This clogs my apps after a while If I have made many mistakes. 

This was a known bug. But now, this is happening with 'app type' variables.  these usually go away but aren't anymore in one of my projects. 

i tried copying the screen and only the components but they arent going away. 

### Steps to Reproduce
1. create a function with input parameters
2. use those variables created by the input parameters somewhere
3. delete the function
4. copy the screen to a new app

### Project Link
https://x.thunkable.com/copy/03f705f72920cb703614d8891604844f

### Expected Behavior
function scope and app scope variables should not still be in the blocks tree. 

### Actual Behavior
I can still call these variables. (i honestly don't even know if they work, i just wanted them gone)
```

<img width="1147" alt="Screen Shot 2020-04-20 at 12 36 41 PM" src="https://user-images.githubusercontent.com/58823942/79776520-dfbbf880-8303-11ea-8704-65c52f463c79.png">

```

**Comment by easaw** [comment]
I've found this too and it's very irritating because when you use the new Screen import, you get the old variables from that screen, which, if you happen to have some of the same name, double up.


Which is bad because it can cause huge issues as multiple screens could have different copies of the same var, where this wouldn't normally happen in a manually coded app. I think these are all things that need to be fixed ASAP please.

**Comment by jaredgib** [comment]
Exactly! My main project example app had that issue but more so when I copy from that to other projects.

**Comment by danielcoglitore** [comment]
me too!

**Comment by domhnallohanlon** [comment]
Hey @easaw @danielcoglitore, this is actually part of the Blocky defaults and will take some work to fix. 

Thanks for flagging this!
